software/ADPL_electron/ADPL_electron.ino

```diff
@@ -127,3 +121,23 @@ int read_temp(int temp_count) {
 void bucket_tipped() {
     bucket.tipped();
 }
+
+
+int publish_data(int last_publish_time) {
+    bool publish_success;
+    char data_str [69];
+
+    sprintf(data_str,"HXCI:%.1f,HXCO:%.1f,HTR:%.1f,HXHI:%.1f,HXHO:%.1f,V:%d,B:%d",
```

Comment:
So currently this will publish a bucket tip count every 2.5 minutes no matter what (even if there have been no tips in the last 2.5mins e.g. `bucket.tip_count==0`). I can easily ignore publishes with a bucket-tip count of 0 on the server, but since bucket tips occur in infrequent batches (relative to when temps are sent) we can save 6-7 bytes on most publishes if we so desire. :).

Though it's not perfect, since `sprintf` ignores extra arguments, you can simply switch format strings on the tip count:
```C++
const char* fmt_string = "HXCI:%.1f,HXCO:%.1f,HTR:%.1f,HXHI:%.1f,HXHO:%.1f,V:%d,B:%d";
const char* fmt_string_no_bucket = "HXCI:%.1f,HXCO:%.1f,HTR:%.1f,HXHI:%.1f,HXHO:%.1f,V:%d";

sprintf(data_str,
    (bucket.tip_count > 0) ? fmt_string : fmt_string_no_bucket,
    tempHXCI.temp, tempHXCO.temp, tempHTR.temp, tempHXHI.temp, tempHXHO.temp,
    int(valve.gasOn), int(bucket.tip_count));
```

More succinctly the ternary condition could be `(bucket.tip_count)`, but perhaps it's more readable the other way?

What do you think?